.github/workflows/upload-release-to-plugin-repo.yml

```diff
@@ -5,10 +5,58 @@ on:
         types: [published]
 
 jobs:
+    compute-should-update-trunk:
+        name: Decide if trunk or tag
+        runs-on: ubuntu-latest
+        outputs:
+            should_update_trunk: ${{ steps.compute_should_update_trunk.outputs.should_update_trunk }}
+
+        steps:
+            - name: Fetch latest version in the WP core repo
+              id: compute_latest_version_in_core_repo
+              run: |
+                  latest_version_in_core_repo=$(curl -s 'https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request\[slug\]=gutenberg' | jq -r '.version')
+                  echo "Latest Core Repo version: $latest_version_in_core_repo"
+                  echo "version=$latest_version_in_core_repo" >> $GITHUB_OUTPUT
+
+            - name: Decide if it is a trunk or tag update
+              id: compute_should_update_trunk
+              env:
+                  GITHUB_REF: ${{ github.ref }}
+              run: |
+                  latestPublishedVersion=$(echo "$GITHUB_REF" | sed -E 's/refs\/tags\/(v?)([0-9.]+)/\2/')
```

Comment:
Where does the ref come from, and how does that connect with the latest version??

Reply:
This workflow is triggered by publishing a draft release. This is the ref from the draft release, hence the name `lastPublishedVersion` for the variable -- it is the version (ref) of the last published release (previously-draft) created by the build zip workflow.